cmd/external-repos/main.go

```diff
@@ -315,82 +315,81 @@ func enqueueSnapshotsCleanup(ctx context.Context, olderThanDays int) error {
 		return fmt.Errorf("error getting new task queue: %w", err)
 	}
 	c := client.NewTaskClient(&q)
-	fs, err := feature_service_client.NewFeatureServiceClient()
-	if err != nil {
-		return fmt.Errorf("error getting feature service client: %w", err)
-	}
-
-	repoConfigDao := dao.GetRepositoryConfigDao(db.DB, pulp_client.GetPulpClientWithDomain(""), fs)
-	snapshotDao := dao.GetSnapshotDao(db.DB)
-	taskInfoDao := dao.GetTaskInfoDao(db.DB)
-
-	repoConfigs, err := repoConfigDao.ListReposWithOutdatedSnapshots(ctx, olderThanDays)
+	daoReg := dao.GetDaoRegistry(db.DB)
+	repoConfigs, err := daoReg.RepositoryConfig.ListReposWithOutdatedSnapshots(ctx, olderThanDays)
```

Comment:
Not part of your changes, but there should be a `defer q.Close()` for the queue here

Reply:
good catch!